src/Hub.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure;
+
+use Symfony\Component\Mercure\Token\TokenProviderInterface;
+
+/**
+ * @author Saif Eddin Gmati <azjezz@protonmail.com>
+ *
+ * @experimental
+ */
+final class Hub
+{
+    private $endpoint;
+    private $provider;
+
+    public function __construct(string $endpoint, TokenProviderInterface $provider)
```

Comment:
```suggestion
    public function __construct(string $url, TokenProviderInterface $jwtProvider)
```

(to be in sync with the terminology we use everywhere).